src/YarpUtilities/include/BipedalLocomotion/YarpUtilities/RosPublisher.h

```diff
@@ -0,0 +1,140 @@
+/**
+ * @file RosPublisher.h
+ * @authors Prashanth Ramadoss
+ * @copyright 2019 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_YARP_UTILITIES_ROS_PUBLISHER_H
+#define BIPEDAL_LOCOMOTION_YARP_UTILITIES_ROS_PUBLISHER_H
+
+#include <BipedalLocomotion/ParametersHandler/IParametersHandler.h>
+#include <BipedalLocomotion/GenericContainer/Vector.h>
+#include <memory>
+#include <iostream>
+#include <string>
+namespace BipedalLocomotion
+{
+
+/**
+ * Helper for YARP library.
+ */
+namespace YarpUtilities
+{
+
+class RosPublisher 
+{
+public:
+    /**
+     * Constructor
+     * @param[in] nodeName Name of the ROS publisher node 
+     */
+    RosPublisher(const std::string& nodeName);
+    
+    /**
+     * Destructor     
+     */
+    ~RosPublisher();
+    
+    /**
+     * Configures the publishers and ROS topics for publishing messages
+     * @param[in] handler Parameter handler
+     * @note this method needs to be called after construction and before publishing  
+     */
+    bool initialize(std::weak_ptr<BipedalLocomotion::ParametersHandler::IParametersHandler> handler);
+    
+    /**
+     * Clear internal configuration and close the node
+     */
+    void stop();
+        
+    /**
+     * Publish the joint states over the configured joint states topic 
+     * The joint velocities and joint efforts are set to zero.
+     * @param[in] jointList list of joints
+     * @param[in] jointPositions list of joint positions in m or rad with the same size as the joints list
+     */
+    bool publishJointStates(const BipedalLocomotion::GenericContainer::Vector<std::string>& jointList,
```

Comment:
Can you try to substitute 
```
const BipedalLocomotion::GenericContainer::Vector<something>&
```
with 
```
const BipedalLocomotion::GenericContainer::Vector<something>::Ref
```
?
This would simplify the usage, as you don't need to use ``make_vector`` before, and you can directly input your original vector. Check https://github.com/dic-iit/bipedal-locomotion-framework/pull/102 and https://github.com/dic-iit/bipedal-locomotion-framework/pull/103.

This applies also everywhere else.

Reply:
Thanks for the suggestion. I knew something was missing there but couldn't point to this.

Reply:
I just updated the comment above for a clarification about ``const``.

Reply:
Fixed in https://github.com/dic-iit/bipedal-locomotion-framework/pull/156/commits/0df2ac4e136da996196f4bf12446664b4c0dd9b7